### PR TITLE
Implemented Keyboard util.

### LINKED ISSUE #20: Implement Keyboard util
With features like

* known key names object,
* parse keystroke pattern,
* KeystrokeInfo interface,
* getCode( key name / keystroke )

### PR DESCRIPTION
**Pull request by Reinmar**
Fixes #20.

### CODE REVIEW

File: src/keyboard.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ *
+ * * `a-z`,
+ * * `0-9`,
+ * * `arrow(left|up|right|bottom)`,
+ * * `backspace`, `delete`, `enter`,
+ * * `ctrl`, `cmd`, `shift`, `alt`.
+ *
+ * @member {Object} utils.keyboard.knownKeyNames
+ */
+export const knownKeyNames = generateKnownKeyNames();
```
Comment (pjasiun) [inline src/keyboard.js]: Is `known` necessary? We do not have variables with arrays of unknown codes ;)

Comment (Reinmar) [inline src/keyboard.js]: Hm... It underlines that these are not all key names. But ok, I'll remove the word and improve the documentation.

**Comment by Reinmar** [comment]
I've just recalled that the error docs are missing. I'll add them as well.

File: src/keyboard.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * Converts a key name or a {@link utils.keyboard.KeystrokeInfo keystroke info} into a key code.
+ *
+ * Note: Key names are matched with {@link utils.keyboard.knownKeyNames} in a case-insensitive way.
+ *
+ * @method utils.keyboard.getCode
+ * @param {String|utils.keyboard.KeystrokeInfo} Key name (see {@link utils.keyboard.knownKeyNames})
+ * or a keystroke data object.
+ * @returns {Number} Key or keystroke code.
+ */
+export function getCode( key ) {
```
Comment (pjasiun) [inline src/keyboard.js]: An example would be useful. It is not obvious what the format of the `key` is using only a docs.

**Comment by Reinmar** [comment]
> I've just recalled that the error docs are missing. I'll add them as well.

Done.

File: src/keyboard.js (hunk 2 above)
Comment (Reinmar) [inline src/keyboard.js]: For now you've got tests. It's not the time to polish the docs.

**Pull request merged**